lib/linter/linter.js

```diff
@@ -1608,25 +1691,109 @@ class Linter {
+                        loc: node.loc,
+                        severity: 1
+                    }));
+
+                });
+            } else {
+                const inlineConfigResult = sourceCode.applyInlineConfig();
+
+                inlineConfigProblems.push(...inlineConfigResult.problems.map(createLintingProblem));
+
+                // need to verify rule existence options
+                if (inlineConfigResult.ok) {
+
+                    const ruleValidator = new RuleValidator();
+
+                    for (const { config: inlineConfig, node } of inlineConfigResult.configs) {
+
+                        Object.keys(inlineConfig.rules).forEach(ruleId => {
+                            const rule = getRuleFromConfig(ruleId, config);
+                            const ruleValue = inlineConfig.rules[ruleId];
+
+                            if (!rule) {
+                                inlineConfigProblems.push(createLintingProblem({ ruleId, loc: node.loc }));
+                                return;
+                            }
+
+                            try {
+
+                                const ruleOptions = Array.isArray(ruleValue) ? ruleValue : [ruleValue];
+
+                                assertIsRuleOptions(ruleId, ruleValue);
+                                assertIsRuleSeverity(ruleId, ruleOptions[0]);
+
+                                ruleValidator.validate({
+                                    plugins: config.plugins,
+                                    rules: {
+                                        [ruleId]: ruleOptions
+                                    }
+                                });
+                                mergedInlineConfig.rules[ruleId] = ruleValue;
+                            } catch (err) {
+
+                                let baseMessage = err.message.slice(
+                                    err.message.startsWith("Key \"rules\":")
+                                        ? err.message.indexOf(":", 12) + 1
+                                        : err.message.indexOf(":")
+                                ).trim();
+
+                                if (err.messageTemplate) {
+                                    baseMessage += ` You passed "${ruleValue}".`;
+                                }
+
+                                inlineConfigProblems.push(createLintingProblem({
+                                    ruleId,
+                                    message: `Inline configuration for rule "${ruleId}" is invalid:\n\t${baseMessage}\n`,
+                                    loc: node.loc
+                                }));
+                            }
+                        });
+                    }
+                }
+            }
+        }
+
         const commentDirectives = options.allowInlineConfig
```

Comment:
> At some point we should refactor how we determine whether inline config should be used, ignored, or warned about, options.allowInlineConfig = true when inline config is not actually allowed but should be warned about is very confusing.

Yeah, things are super confusing around this behavior. I think it would be best to leave as-is and look at addressing in the rewrite.